### DISCUSSION TITLE
Clarification about memory order

### DISCUSSION THREAD

**Discussion by wisp3rwind** [dimforge/nalgebra#1225] [Q&A] [answered]
The docs state the following things on memory-order

- [`Matrix` is "The most generic column-major matrix (and vector) type."](https://docs.rs/nalgebra/latest/nalgebra/base/struct.Matrix.html)
- [`MatrixXx3` is column-major](https://docs.rs/nalgebra/latest/nalgebra/base/type.MatrixXx3.html)
- [`Matrix3xX` is **row**-major](https://docs.rs/nalgebra/latest/nalgebra/base/type.Matrix3xX.html) (but it is also an alias of `Matrix<...>`)

Is this really correct? My current (admittedly superficial) understanding of `nalgebra` is that it only supports a single memory layout, which is column-major. But then, `Matrix3xX` should be column-major, too. If not, how would I build a 3xX column-major matrix?

Thanks!

**Comment by sebcrozet** [comment, marked as answer]
You are correct, `Matrix3xX`, like all the other matrices, is column-major.
